lib/postgrex/replication.ex

```diff
@@ -354,28 +354,35 @@ defmodule Postgrex.Replication do
   end
 
   @doc """
-  Starts logical replication on the given slot. If the slot's plugin requires
-  additional options, make sure to specify them using the `plugin_opts` option.
+  Starts logical replication on the given slot.
+
+  If the slot's plugin requires additional options, the `plugin_opts`
+  option must be specified.
 
   If the connection was started with `auto_reconnect` set to `true`, then
   replication will automatically restart with the options passed into this
   function. You must ensure your system will not be affected by receiving
-  duplicate WAL updates. Note that temporary slots cannot be restarted due to the
-  fact that they automatically drop upon disconnect.
+  duplicate WAL updates. Restarting temporary slots requires the `plugin`
+  option in order to recreate the slot.
 
   This function will return `{:error, :stream_in_progress}` while a table
   is being copied or after replication has begun.
 
   ## Options
 
-    * `:plugin_opts` - It must be a keyword list and is used to configure
-      the output plugin assigned to the given slot.
-
     * `:start_pos` - The LSN value to start replication from. Must be
       formatted as a string of two hexadecimal numbers of up to 8 digits
       each, separated by a slash. e.g. `1/F73E0220`.
       Defaults to `0/0`.
 
+    * `:plugin_opts` - It must be a keyword list and is used to configure
+      the output plugin assigned to the given slot.
+
+    * `:plugin` - Used to recreate temporary slots when `auto_reconnect = true`.
+      If not provided, the slot cannot be recreated and an error will be raised
+      during reconnection. Recreated slots will be made with `temporary = true`
+      and `snapshot = :noexport`. See `create_slot/4` for more details.
```

Comment:
I am not sure if we should couple the plugin with the temporary option. Someone may want to use plugin for other reasons without setting temporary or snapshot on reconnect?

Reply:
Now that you mention this, I am realizing we can reconnect temporary slots without  adding new options using the results from [command(:slot_state, opts)](https://github.com/elixir-ecto/postgrex/blob/master/lib/postgrex/replication.ex#L1017). This will tell us the plugin and whether the slot is temporary or not, given the slot name. 

If you like this route, there would be a couple questions:

1. We can't receive the `:snapshot` value from this command. I'm not sure a user will be able to act on an exported snapshot when reconnecting, so I think it could make sense to set this as `:noexport`.
2. Should permanent slots be recreated on reconnect or just temporary ones? I'm thinking they shouldn't because if a permanent slot is disappearing between disconnect and reconnect probably the user would want to know. 

Anyway, those are just my initial thoughts. Please let me know what you think.

Reply:
My suggestion would be this:

1. If `temporary: true`, we recreate. If we also need the plugin option, you should raise if temporary+auto_reconnect are given and there is no plugin. If a snapshot is also required, then we will request it to be added too.

2. If `temporary: false`, we don't recreate.

This way the user intent is always clear. WDYT?

Reply:
Perfect! I will make the change.

Reply:
I made the changes but just wanted to double check with you a couple things:

1. I validate before starting replication just so the user doesn't find a surprise when they reconnect after a few days. Wanted to make sure that is in line with how you would normally do things.
2. Is there a good way to test that an exception is raised inside the connection process? I tried a few things, `assert_raise` and trapping exits but couldn't find a way to get it to work.

Reply:
You should be able to trap exits and then do assert_receive on the EXIT message that gets delivered to your inbox.

regarding the raising, perfect.

Reply:
Ah I was missing `catch_exit` when I tried to trap exits before. Working now. Thank you for the help.